ratis-server/src/main/java/org/apache/ratis/server/impl/RaftServerImpl.java

```diff
@@ -1604,14 +1604,21 @@ private InstallSnapshotReplyProto notifyStateMachineToInstallSnapshot(
                     LOG.debug("{}: StateMachine could not install snapshot as it is not available", this);
                   }
                 }
-              // wait for 10 seconds for statemachine to install snapshot
+              // wait for 1 seconds for statemachine to install snapshot
               }).get(1, TimeUnit.SECONDS);
-        } catch (ExecutionException t) {
+        } catch (InterruptedException | TimeoutException t) {
+          //nothing to do
+        } catch (Exception t) {
+          // there are two cases:
+          //1 `get()` may throw ExecutionException if `whenComplete` throw an exception
+          //2 when generating completeFuture, `statemachine#notifyInstallSnapshotFromLeader`
+          // may throw an uncertain exception, which is determined by the implementation of
+          // user statemachine.
           inProgressInstallSnapshotRequest.compareAndSet(firstAvailableLogIndex, 0);
-          LOG.warn("{}: Failed to notify StateMachine to InstallSnapshot. ExecutionException: {}",
+          LOG.warn("{}: Failed to notify StateMachine to InstallSnapshot. Exception: {}",
               getMemberId(), t.getMessage());
           throw new IOException("Failed to install snapshot");
```

Comment:
Let's add the exception t to the IOException.
```
          final String err = getMemberId() + ": Failed to notify StateMachine to InstallSnapshot.";
          LOG.warn(err + " " + t);
          throw new IOException(err, t);
```

Reply:
thanks @szetszwo for the review. i have updated this patch , please take a look